File: 779C/dishonest_sellers.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

typedef pair<int, int> ii;
typedef vector<int> vi;
typedef vector<pair<int, int>> vii;

int main() {
  int n, k, nc;
  cin >> n >> k;

  vi al;
  nc = n;
  while (nc > 0) {
    int a;
    cin >> a;
    al.push_back(a);
    --nc;
  }

  vi bl;
  nc = n;
  while (nc > 0) {
    int b;
    cin >> b;
    bl.push_back(b);
    --nc;
  }

  vii diff;
  for (int i = 0; i < n; ++i) {
    ii d = {al[i] - bl[i], i};
    diff.push_back(d);
  }
  
  sort(begin(diff), end(diff), [](auto &a, auto &b) {
    return a.first < b.first;
  });

  int total = 0;
  int i = 0, min_index;

  while (k > 0) {
    min_index = diff[i].second;
    total += al[min_index];
    al[min_index] = 0;
    bl[min_index] = 0;
    ++i;   
    --k;  
  }

  while (i < n && diff[i].first < 0) {
    min_index = diff[i].second;
    total += al[min_index];
    al[min_index] = 0;
    bl[min_index] = 0;
    ++i;   
  }

  for (auto value : bl) {
    total += value;
  }

  cout << total << endl;

  return 0;
}

File: 779C/dishonest-sellers.py
""" Created by Henrikh Kantuni on 2/26/17 """

if __name__ == "__main__":
    n, k = [int(x) for x in input().split()]
    al = [int(x) for x in input().split()]
    bl = [int(x) for x in input().split()]

    diff = []
    for i in range(n):
        diff.append((al[i] - bl[i], i))

    diff.sort(key=lambda x: x[0])

    total = 0
    while k > 0:
        min_index = diff[0][1]
        total += al[min_index]
        al[min_index] = 0
        bl[min_index] = 0
        del diff[0]
        k -= 1

    while diff and diff[0][0] < 0:
        min_index = diff[0][1]
        total += al[min_index]
        al[min_index] = 0
        bl[min_index] = 0
        del diff[0]

    for v in bl:
        total += v

    print(total)
